app/scripts/controllers/serviceInstance.js

```diff
@@ -37,25 +58,68 @@ angular.module('openshiftConsole')
 
-      ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function(serviceClass) {
-        $scope.serviceClass = serviceClass;
-        $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, serviceClass);
-        updateBreadcrumbs();
+      $scope.parameterData = {};
+      _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function(key) {
+        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      });
+
+      $scope.parameterData = angular.extend($scope.parameterData, _.get($scope.serviceInstance, 'spec.parameters', {}));
+
+      _.each(_.get($scope.serviceInstance, 'spec.parametersFrom'), function(parametersSource) {
+        secretWatcher = DataService.watchObject("secrets", _.get(parametersSource, 'secretKeyRef.name'), $scope.projectContext, function (secret) {
+          try {
+            _.extend($scope.parameterData, JSON.parse(SecretsService.decodeSecretData(secret.data).parameters));
+          } catch (e) {
+          }
+        });
       });
     };
 
-    var updatePlan = function() {
-      if (ServiceInstancesService.isCurrentPlan($scope.serviceInstance, $scope.plan)) {
+    var updateParameterSchema = function() {
+      $scope.parameterFormDefinition = angular.copy(_.get($scope.plan, 'spec.externalMetadata.schemas.service_instance.create.openshift_form_definition'));
+
+      $scope.parameterSchema = angular.copy(_.get($scope.plan, 'spec.instanceCreateParameterSchema'));
+      $scope.parameterSchema.readonly = true;
+      $scope.parameterSchema.required = [];
+
+      _.each(_.get($scope.parameterSchema, 'properties'), function(property) {
+        property.description = undefined;
+        property.title = property.title + ":";
+      });
+
+      // TODO: Find some plans and schemas that are updateable
+      // var planUpdatable = _.get($scope.plan, 'spec.instanceUpdateParameterSchema') || (_.get($scope.serviceClass, 'spec.planUpdatable') && (_.size($scope.servicePlans) > 1));
+      var planUpdatable = _.get($scope.plan, 'spec.instanceCreateParameterSchema') || (_.get($scope.serviceClass, 'spec.planUpdatable') && (_.size($scope.servicePlans) > 1));
+
+      $scope.editAvailable = planUpdatable && serviceInstanceReady($scope.serviceInstance) && !_.get($scope.serviceInstance, 'metadata.deletionTimestamp');
+    };
+
+    var updateServiceClass = function() {
+      if(!$scope.serviceInstance || !$scope.serviceClasses) {
         return;
       }
 
-      ServiceInstancesService.fetchServicePlanForInstance($scope.serviceInstance).then(function(plan) {
-        $scope.plan = plan;
+      var serviceClassName = _.get($scope.serviceInstance.spec, 'serviceClassRef.name');
+      $scope.serviceClass = _.get($scope.serviceClasses, [serviceClassName]);
+      $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, $scope.serviceClass);
+      updateBreadcrumbs();
+
+      Catalog.getServicePlans().then(function(plans) {
```

Comment:
We should be able to only request plans for a specific service class, but OK for now. I worry down the road that this might be a lot of data, though.